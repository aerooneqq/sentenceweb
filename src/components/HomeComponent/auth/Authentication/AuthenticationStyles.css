#authenticationContainer {
    background-color: transparent;
    margin-top: 20px;
    border-style: solid;
    border-radius: 5px;
    border-width: 0px;
    border-color: lightgray;
    z-index: 2;
}

.authenticationContainerSize{ 
  width: 400px;
  height: 450px;
}

@media (max-width: 500px){ 
  .authenticationContainerSize{ 
    width: 100%;
    height: 450px;
  }
}

#signModuleContainer{
  width: 100%;
  height: 100%;

  padding-top: 20px;
  margin-top: 10px;

  border-radius: 5px;
  background: #f0f0f0;
}

.rotate{
  transform: rotateY(180deg);
  -moz-transform: rotateY(180deg);
  -webkit-transform: rotateY(180deg);
  -o-transform: rotateY(180deg);
}

#rotationContainer, #front, #back {
  -moz-perspective: 1000;
  -webkit-perspective: 1000;
  perspective: 1000;


  width: 100%;
  height: 100%;

  margin-top: 10px;
  padding-top:  0px;
  border-radius: 5px;
  background: #f0f0f0;
}

#rotation{
  -moz-transition: 1s;
  -webkit-transition: 1s;
  transition: 1s;

  transform-style: preserve-3d;
  -moz-transform-style: preserve-3d;

  position: relative;
}

#back, #front{
  transform: rotateX(0deg);

  backface-visibility: hidden;
  -moz-backface-visibility: hidden;
  -webkit-backface-visibility: hidden;
  -moz-backface-visibility: hidden;

  position: absolute;
  top: 0;
  left: 0;
}

#back{
  transform: rotateY(180deg);
  -moz-transform: rotateY(180deg);
  -webkit-transform: rotateY(180deg);
}

#front{
  z-index: 2;
}
