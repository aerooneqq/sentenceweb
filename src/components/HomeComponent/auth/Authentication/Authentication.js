import React, {Suspense, lazy} from "react"

//Styles
import "./AuthenticationStyles.css"

//Components
const AuthenticationTopBar = lazy(() => import("../AuthenticationTopBar/AuthenticationTopBar"))
const Authorization = lazy(() => import("../Authorization/Authorization.js"))
const Registration = lazy(() => import("../Registration/Registration.js"))
const DeviceContext = lazy(() => import("../../../../contexts/DeviceContext"))

export default class Authentication extends React.Component {
    static contextType = DeviceContext

    constructor(props) {
        super(props)

        this.state = {
          mode: "signIn"
        }

        this.changeSignMode = this.changeSignMode.bind(this)
    }

    /**
     * Changes the mode of this component. 
     * The change comes from the AuthenticationTopBar component.
     * Mode can take two values: "signIn" and "signUp". When the change of the mode is performed,
     * the animation of card rotation is applied to the authentication block
     */
    changeSignMode() {
      document.getElementById("rotation").classList.toggle("rotate");

      this.setState(state => { 
        return { 
          mode: state.mode === "signIn" ? "signUp" : "signIn"
        };
      });
    }

    render() {
        return (
          <Suspense fallback = {<div>Loading...</div>}>
            <div id="authenticationContainer" class = "authenticationContainerSize">
                <AuthenticationTopBar mode = {this.state.mode}
                                      changeSignMode = {this.changeSignMode}/>
                <div id ="rotationContainer">
                  <div id="rotation">
                    <div id = "front">
                      <Authorization signIn = {this.props.signIn}/>
                    </div>
                    <div id = "back">
                      <Registration />
                    </div>
                  </div>
                </div>
            </div>
          </Suspense>
        )
    }
}
