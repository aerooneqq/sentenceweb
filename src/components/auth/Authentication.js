import React, {Suspense, lazy} from "react"
import "./AuthenticationStyles.css"

const AuthenticationTopBar = lazy(() => import("./AuthenticationTopBar"))
const Authorization = lazy(() => import("./Authorization.js"))
const Registration = lazy(() => import("./Registration.js"))
const DeviceContext = lazy(() => import("../../contexts/DeviceContext"))

export default class Authentication extends React.Component {
    static contextType = DeviceContext

    constructor(props) {
        super(props)

        this.state = {
          mode: "signIn"
        }

        this.changeSignMode = this.changeSignMode .bind(this)
    }

    changeSignMode(){
      let rotatableCont = document.getElementById("rotation")
      rotatableCont.classList.toggle("rotate")

      if (this.state.mode === "signIn"){
        this.setState({
          mode: "signUp"
        })
      }
      else {
        this.setState({
          mode: "signIn"
        })
      }
    }

    render() {
        return (
          <Suspense fallback = {<div>Loading...</div>}>
            <div id="authenticationContainer" className = {this.context === "mobile" ?
                  "mobileAuthenticationContainer" : "desktopAuthenticationContainer"}>
                <AuthenticationTopBar mode = {this.state.mode} changeSignMode
                    = {this.changeSignMode}/>

                <div id ="rotationContainer">
                  <div id="rotation">
                    <div id="front">
                      <Authorization/>
                    </div>
                    <div id = "back">
                      <Registration />
                    </div>
                  </div>
                </div>

            </div>
          </Suspense>
        )
    }
}
