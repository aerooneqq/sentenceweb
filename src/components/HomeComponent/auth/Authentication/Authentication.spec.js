import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import Authentication from './Authentication';


describe("Authentiation top bar", () => { 
    it("should render correctly", () => { 
        const props = { 
            signIn: () => {}
        }

        const authentication = shallow(<Authentication {...props} />);

        expect(shallowToJson(authentication)).toMatchSnapshot();
    });  

    it("should change state", () => { 
        const props = { 
            signIn: () => {}
        }

        const authentication = shallow(<Authentication {...props}/>);

        authentication.instance().changeSignMode();
        expect(authentication.state().mode).toBe("signUp");
    });
});