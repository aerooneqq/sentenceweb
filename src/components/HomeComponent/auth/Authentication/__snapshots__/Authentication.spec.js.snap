// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentiation top bar should render correctly 1`] = `
<Suspense
  fallback={
    <div>
      Loading...
    </div>
  }
>
  <div
    class="authenticationContainerSize"
    id="authenticationContainer"
  >
    <lazy
      changeSignMode={[Function]}
      mode="signIn"
    />
    <div
      id="rotationContainer"
    >
      <div
        id="rotation"
      >
        <div
          id="front"
        >
          <lazy
            signIn={[Function]}
          />
        </div>
        <div
          id="back"
        >
          <lazy />
        </div>
      </div>
    </div>
  </div>
</Suspense>
`;
